packages/hierarchy-builder/src/test/HierarchyNode.test.ts

```diff
@@ -156,6 +160,334 @@ describe("HierarchyNodeKey", () => {
       expect(HierarchyNodeKey.equals(baseValueRange, { ...baseValueRange, propertyClassName: "Schema.Other" })).to.be.false;
     });
   });
+
+  describe("compare", () => {
+    describe("key types are different", () => {
+      const customNodeKey = "x";
+      const instanceNodeKey: InstancesNodeKey = { type: "instances", instanceKeys: [] };
+      const classGroupingNodeKey: ClassGroupingNodeKey = { type: "class-grouping", class: { name: "x" }, groupedInstanceKeys: [] };
+      const labelGroupingNodeKey: LabelGroupingNodeKey = { type: "label-grouping", label: "a", groupedInstanceKeys: [] };
+      const propertyGroupingOtherNodeKey: PropertyOtherValuesGroupingNodeKey = { type: "property-grouping:other", groupedInstanceKeys: [] };
+      const propertyGroupingValueNodeKey: PropertyValueGroupingNodeKey = {
+        type: "property-grouping:value",
+        propertyClassName: "",
+        propertyName: "",
+        formattedPropertyValue: "",
+        groupedInstanceKeys: [],
+      };
+      const propertyGroupingRangeNodeKey: PropertyValueRangeGroupingNodeKey = {
+        type: "property-grouping:range",
+        propertyClassName: "",
+        propertyName: "",
+        fromValue: 1,
+        toValue: 2,
+        groupedInstanceKeys: [],
+      };
+
+      it("returns correct results when one of the keys is an instance node key", () => {
```

Comment:
Lots of compares here, a bit hard to read :) What do you think about the following approach:
1. define a single array with keys of different types
2. ```ts
    keys.forEach((lhs, lhsIndex) => {
      keys.forEach((rhs, rhsIndex) => {
        const lhsToRhs = HierarchyNodeKey.compare(lhs, rhs);
        const rhsToLhs = HierarchyNodeKey.compare(rhs, lhs);
        if (lhsIndex === rhsIndex) {
          expect(lhsToRhs).to.eq(0);
          expect(rhsToLhs).to.eq(0);
        } else {
          expect(lhsToRhs).to.not.eq(0);
          expect(rhsToLhs).to.not.eq(0);
          expect(rhsToLhs).to.eq(-1 * lhsToRhs);
        }
      });
    });
    ```

Reply:
Yeah I agree, though about doing this, but for some reason decided that the original implementation was simpler to understand. 
Changed to your suggestion.